### PR TITLE
resources: support for new OverlayPlugin log lines

### PR DESCRIPTION
**Pull request by wexxlee**
_(edited 03-02-2024, 03-03-2024)_
- Adds regex/netregex support for `ContentFinderSettings` (0x109)
- Adds definitions, regex/netregex support, and unit tests for `NpcYell` (0x10A), `BattleTalk2` (0x10B), `Countdown` (0x10C), `CountdownCancel` (0x10D), `ActorMove` (0x10E), `ActorSetPos` (0x10F), `SpawnNpcExtra` (0x110), and `ActorControlExtra` (0x111).
- Updates the log guide to include the previous eight log lines.

NB: This PR should be merged after OP bumps to v0.19.27, which will include the new log lines.

### CODE REVIEW

**Comment by valarnin** [comment]
(for future reference, the auto lint PR title script allows `[WIP]` as a valid prefix, e.g. the title of this could have been `[WIP]  resources: support for new OverlayPlugin log lines`)

File: docs/LogGuide.md [hunk 1]
```diff
@@ -2821,3 +2853,399 @@ Parsed Log Line Examples:
 ```
 
 <!-- AUTO-GENERATED-CONTENT:END (logLines:type=ContentFinderSettings&lang=en-US) -->
+
+<a name="line266"></a>
+
+### Line 266 (0x10A): NpcYell
+
+This log line is emitted whenever a NpcYell packet is received from the server,
+indicating that an NPC has yelled something (e.g. UWU Nael quotes).
```
Comment (valarnin) [inline docs/LogGuide.md:2862]:
```suggestion
indicating that an NPC has yelled something (e.g. UCOB Nael quotes).
```

File: docs/LogGuide.md [hunk 2]
```diff
@@ -2821,3 +2853,399 @@ Parsed Log Line Examples:
...
+<!-- AUTO-GENERATED-CONTENT:END (logLines:type=BattleTalk2&lang=en-US) -->
+
+<a name="line268"></a>
+
+### Line 268 (0x10C): Countdown
+
+This log line is emitted whenever a countdown is started.
+
+`result` is `00` if successful, and non-zero if the attempt to start a countdown failed
+(e.g., if a countdown is already in progress, or if combat has started).
```
Comment (valarnin) [inline docs/LogGuide.md:2963]: It may be worth it to note here that there is no network packet for the countdown finishing, and therefore no line will be (or can reasonably be) emitted for the `Engage!` log line.

File: docs/LogGuide.md [hunk 3]
```diff
@@ -2821,3 +2853,399 @@ Parsed Log Line Examples:
...
+<!-- AUTO-GENERATED-CONTENT:END (logLines:type=CountdownCancel&lang=en-US) -->
+
+<a name="line270"></a>
+
+### Line 270 (0x10E): ActorMove
+
+An `ActorMove` packet is sent to instruct the game client to move an actor to a new position
+(for example, in UWU, when Titan turns prior to jumping to indicate the direction of his jump).
+The FFXIV client interpolates the actor's movement between the current position and the new position.
+Currently, these log lines are emitted only for non-player actors (id >= 0x40000000).
```
Comment (valarnin) [inline docs/LogGuide.md:3050]: I think this is a bit misleading? I'd rephrase it something like `... to a new position whenever they have been moved. This can be used to detect rapid movement which would otherwise be lost, for example, in UWU, when...`

File: docs/LogGuide.md [hunk 4]
```diff
@@ -2821,3 +2853,399 @@ Parsed Log Line Examples:
...
+```
+
+<!-- AUTO-GENERATED-CONTENT:END (logLines:type=ActorMove&lang=en-US) -->
+
+<a name="line271"></a>
+
+### Line 271 (0x10F): ActorSetPos
+
+An `ActorSetPos` packet is sent to instruct the game client to set the position of an actor
+with no interpolated movement (for example, in UWU, to set the location of the Ifrit clones).
```
Comment (valarnin) [inline docs/LogGuide.md:3095]: It may be worth noting that sometimes these `ActorSetPos` lines are accompanied by other data (in other log lines or network packets) indicating that an animation should be played if the actor is already visible.

This could for example be something like:
1. Ability use with animation tied to it, for example to make the actor appear to be jumping
2. A few seconds later, `ActorSetPos` is sent to move actor
3. Ability use or other network packet to indicate an animation should be played, for example to make the actor appear to be landing

File: docs/LogGuide.md [hunk 5]
```diff
@@ -2821,3 +2853,399 @@ Parsed Log Line Examples:
...
+This line contains certain data from `NpcSpawn` packets not otherwise made available
+through other log lines.
+
+The `tetherId` field is the same as the `id` field used in
+[NetworkTether lines](#line-35-0x23-networktether) and corresponds to an id in the
+[Channeling table](https://github.com/xivapi/ffxiv-datamining/blob/master/csv/Channeling.csv).
+
+The `animationState` field reflects the initial animation state of the actor
+at time it is spawned, and corresponds to the
+[BNpcState table](https://github.com/xivapi/ffxiv-datamining/blob/master/csv/BNpcState.csv).
```
Comment (valarnin) [inline docs/LogGuide.md:3148]: Might be worth mentioning the nuance here of "if the actor spawns with a tether or animation state value, there will not be a corresponding `NetworkTether` or `ActorControlExtra` line emitted to indicate this information"?

File: ui/raidboss/data/00-misc/test.ts [hunk 6]
```diff
@@ -365,6 +365,29 @@ const triggerSet: TriggerSet<Data> = {
...
+      type: 'Countdown',
+      netRegex: { result: '00' },
+      infoText: (_data, matches, output) =>
+        output.countdown!({ player: matches.name, seconds: matches.countdownTime }),
+      outputStrings: {
+        countdown: {
+          en: '${player} started ${seconds}s countdown',
+        },
+      },
+    },
```
Comment (valarnin) [inline ui/raidboss/data/00-misc/test.ts:379]: Maybe add an additional test trigger here for `Test Countdown Failed`, where `result` !== `00`, showing the result code?

**Review by valarnin — CHANGES_REQUESTED** [review]

File: docs/LogGuide.md (hunk 1 above)
Comment (wexxlee) [reply docs/LogGuide.md:2862]: 🤦

File: docs/LogGuide.md [hunk 7]
```diff
@@ -2821,3 +2853,420 @@ Parsed Log Line Examples:
...
+
+These log lines are sometimes accompanied by other data (other log lines or network packets)
+indicating that an animation should be played if the actor is visible.
+For example, the following log lines (or network packets) might be sent
+in sequence to have an enemy appear to jump to a new location:
+
+1. A [NetworkAbility line](#line-21-0x15-networkability) line with an associated animation
+   to make it appear as though the actor is jumping.
+2. An `ActorSetPos` line to change the actor's location.
+3. Another `NetworkAbility` line (oir other packet) with an associated animation to make it
```
Comment (valarnin) [inline docs/LogGuide.md:3111]:
```suggestion
3. Another `NetworkAbility` line (or other packet) with an associated animation to make it
```

**Review by valarnin — APPROVED** [review]
This looks good other than a typo.

File: docs/LogGuide.md [hunk 8]
```diff
@@ -2821,3 +2853,420 @@ Parsed Log Line Examples:
...
+
+### Line 268 (0x10C): Countdown
+
+This log line is emitted whenever a countdown is started.
+
+`result` is `00` if successful, and non-zero if the attempt to start a countdown failed
+(e.g., if a countdown is already in progress, or if combat has started).
+
+Note: Because there is no network packet sent when a countdown completes successfully,
+no log line is (or reaosnably can be) emitted for the 'Engage!' message.
```
Comment (xiashtra) [inline docs/LogGuide.md:2966]:
```suggestion
no log line is (or reasonably can be) emitted for the 'Engage!' message.
```

File: docs/LogGuide.md [hunk 9]
```diff
@@ -2821,3 +2853,420 @@ Parsed Log Line Examples:
...
+
+<!-- AUTO-GENERATED-CONTENT:END (logLines:type=CountdownCancel&lang=en-US) -->
+
+<a name="line270"></a>
+
+### Line 270 (0x10E): ActorMove
+
+An `ActorMove` packet is sent to instruct the game client to move an actor to a new position
+whenever they have been moved.
+This can be used, for example, to detect rapid movement which would overwisae be lost
```
Comment (xiashtra) [inline docs/LogGuide.md:3052]:
```suggestion
This can be used, for example, to detect rapid movement which would otherwise be lost
```

File: docs/LogGuide.md [hunk 10]
```diff
@@ -2821,3 +2853,420 @@ Parsed Log Line Examples:
...
+
+An `ActorSetPos` packet is sent to instruct the game client to set the position of an actor
+with no interpolated movement (for example, in UWU, to set the location of the Ifrit clones).
+
+These log lines are sometimes accompanied by other data (other log lines or network packets)
+indicating that an animation should be played if the actor is visible.
+For example, the following log lines (or network packets) might be sent
+in sequence to have an enemy appear to jump to a new location:
+
+1. A [NetworkAbility line](#line-21-0x15-networkability) line with an associated animation
```
Comment (xiashtra) [inline docs/LogGuide.md:3108]:
```suggestion
1. A [NetworkAbility](#line-21-0x15-networkability) line with an associated animation
```

File: docs/LogGuide.md [hunk 11]
```diff
@@ -2821,3 +2853,420 @@ Parsed Log Line Examples:
...
+
+<a name="line272"></a>
+
+### Line 272 (0x110): SpawnNpcExtra
+
+This line contains certain data from `NpcSpawn` packets not otherwise made available
+through other log lines.
+
+The `tetherId` field is the same as the `id` field used in
+[NetworkTether lines](#line-35-0x23-networktether) and corresponds to an id in the
```
Comment (xiashtra) [inline docs/LogGuide.md:3160]:
```suggestion
[NetworkTether](#line-35-0x23-networktether) lines and corresponds to an id in the
```

File: docs/LogGuide.md [hunk 12]
```diff
@@ -2821,3 +2853,420 @@ Parsed Log Line Examples:
...
+[Channeling table](https://github.com/xivapi/ffxiv-datamining/blob/master/csv/Channeling.csv).
+
+The `animationState` field reflects the initial animation state of the actor
+at time it is spawned, and corresponds to the
```
Comment (xiashtra) [inline docs/LogGuide.md:3164]:
```suggestion
at the time it is spawned, and corresponds to the
```

**Review by xiashtra — COMMENTED** [review]
a few typos

**Comment by wexxlee** [comment]
FWIW, I was leaving the suggestions in place and holding this PR open intentionally to try to resolve a continuing bug with approval workflows. :-(

**Comment by ShadyWhite** [comment]
> FWIW, I was leaving the suggestions in place and holding this PR open intentionally to try to resolve a continuing bug with approval workflows. :-(

Sorry, I only thought the OP version had been bumped to 0.19.27 so merged this PR.

**Comment by wexxlee** [comment]
Yeah, my fault, I should have dropped a comment.  All good though, I think I've got a fix figured out.
